src/test/java/org/prebid/server/bidder/intertech/IntertechBidderTest.java

```diff
@@ -0,0 +1,368 @@
+        final BidRequest bidRequest = givenBidRequest(
+                impBuilder -> impBuilder
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode()))),
+                identity());
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = intertechBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors())
+                .allSatisfy(error -> {
+                    assertThat(error.getType()).isEqualTo(BidderError.Type.bad_input);
+                    assertThat(error.getMessage()).startsWith("imp #0: Cannot deserialize value");
+                });
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenPageIdIsEmpty() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                impBuilder -> impBuilder
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpIntertech.of(null, 7)))),
+                identity());
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = intertechBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).containsExactly(BidderError.badInput("imp #0: missing param page_id"));
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenImpIdIsEmpty() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                impBuilder -> impBuilder
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpIntertech.of(123456, 0)))),
+                identity());
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = intertechBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).containsExactly(BidderError.badInput("imp #0: missing param imp_id"));
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldCreateARequestForEachImpAndSkipImpsWithErrors() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(asList(
+                        givenImp(impBuilder -> impBuilder.id("imp1")),
+                        givenImp(impBuilder -> impBuilder.id("imp2")
+                                .ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpIntertech.of(1234567, null))))),
+                        givenImp(impBuilder -> impBuilder.id("imp3"))))
+                .build();
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = intertechBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).containsExactly(BidderError.badInput("imp #1: missing param imp_id"));
+        assertThat(result.getValue()).hasSize(2)
+                .extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
```

Comment:
We have this `payload` field that we utilize in tests, use it instead of unmarshalling.